pallets/creator-staking/src/functions.rs

```diff
@@ -0,0 +1,309 @@
+            GeneralStakerInfo::<T>::insert(staker, creator_id, staker_info)
+        }
+    }
+
+    /// Calculate the reward distribution between a creator and all their staking participants.
+    ///
+    /// Returns (creator's reward, stakers' combined reward)
+    pub(crate) fn distributed_rewards_between_creator_and_stakers(
+        creator_info: &CreatorStakeInfo<BalanceOf<T>>,
+        era_info: &EraInfo<BalanceOf<T>>,
+    ) -> (BalanceOf<T>, BalanceOf<T>) {
+        let creator_proportional_stake =
+            Perbill::from_rational(creator_info.total, era_info.staked);
+
+        let creator_reward_share = creator_proportional_stake * era_info.rewards.creators;
+        let stakers_combined_reward_share = creator_proportional_stake * era_info.rewards.stakers;
+
+        (creator_reward_share, stakers_combined_reward_share)
+    }
+
+    pub(crate) fn account_id() -> T::AccountId {
+        T::PalletId::get().into_account_truncating()
+    }
+
+    /// The block rewards are accumulated on the pallets's account during an era.
+    /// This function takes a snapshot of the pallet's balance accrued during current era
+    /// and stores it for future distribution
+    ///
+    /// This is called just at the beginning of an era.
+    pub(super) fn reward_balance_snapshot(era: EraIndex, rewards: RewardInfo<BalanceOf<T>>) {
+        // Get the reward and stake information for previous era
+        let mut era_info = Self::general_era_info(era).unwrap_or_default();
+
+        // Prepare info for the next era
+        GeneralEraInfo::<T>::insert(
+            era + 1,
+            EraInfo {
+                rewards: Default::default(),
+                staked: era_info.staked,
+                locked: era_info.locked,
+            },
+        );
+
+        // Set the reward for the previous era.
+        era_info.rewards = rewards;
+
+        GeneralEraInfo::<T>::insert(era, era_info);
+    }
+
+    /// Used to copy all `CreatorStakeInfo` from the ending era over to the next era.
+    /// This is the most primitive solution since it scales with number of creators.
+    /// It is possible to provide a hybrid solution which allows laziness but also prevents
+    /// a situation where we don't have access to the required data.
+    pub(super) fn rotate_staking_info(current_era: EraIndex) -> Weight {
+        let next_era = current_era + 1;
+
+        let mut consumed_weight = Weight::zero();
+
+        for (space_id, creator_info) in RegisteredCreators::<T>::iter() {
+            // Ignore creator if it was unregistered
+            consumed_weight = consumed_weight.saturating_add(T::DbWeight::get().reads(1));
+            if let CreatorState::Unregistered(_) = creator_info.state {
+                continue;
+            }
+
+            // Copy data from era `X` to era `X + 1`
+            if let Some(mut staking_info) = Self::creator_stake_info(space_id, current_era)
+            {
+                staking_info.creator_reward_claimed = false;
```

Comment:
Why do we do this?

Reply:
Not to double claim rewards by creator for the specific era.